From 793b1f81f0dc20d41a9efd02771c90edb9cfda10 Mon Sep 17 00:00:00 2001
From: Murat Kozan <muratkozan350@gmail.com>
Date: Fri, 24 Feb 2023 21:34:02 +0300
Subject: [PATCH] init.rc: Don't enable f2fs iostat by default

---
 rootdir/init.rc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 6278350da2da..30514b7665b5 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1084,7 +1084,6 @@ on boot
     # are not aware of using fsync()/sync() to prepare sudden power-cut.
     write /dev/sys/fs/by-name/userdata/cp_interval 200
     write /dev/sys/fs/by-name/userdata/gc_urgent_sleep_time 50
-    write /dev/sys/fs/by-name/userdata/iostat_enable 1
 
     # limit discard size to 128MB in order to avoid long IO latency
     # for filesystem tuning first (dm or sda)
