From 5631c3e6ac4d856a129400a76946bb70b676cd73 Mon Sep 17 00:00:00 2001
From: Astridxx <muratkozan350@gmail.com>
Date: Mon, 12 Feb 2024 17:07:52 +0300
Subject: [PATCH] init: Don't enable f2fs iostat by default

---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index e21468ea02ba..74f16bda3521 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1179,7 +1179,7 @@ on boot
     write /dev/sys/fs/by-name/userdata/cp_interval 200
     write /dev/sys/fs/by-name/userdata/gc_urgent_sleep_time 50
     write /dev/sys/fs/by-name/userdata/iostat_period_ms 1000
-    write /dev/sys/fs/by-name/userdata/iostat_enable 1
+    write /dev/sys/fs/by-name/userdata/iostat_enable 0
 
     # set readahead multiplier for POSIX_FADV_SEQUENTIAL files
     write /dev/sys/fs/by-name/userdata/seq_file_ra_mul 128
