From 570b3379d76ed6a15f0634bf323cc93091e68338 Mon Sep 17 00:00:00 2001
From: Murat Kozan <muratkozan350@gmail.com>
Date: Thu, 8 Feb 2024 17:45:45 +0300
Subject: [PATCH] BluetoothControllerImpl: Optimize connected battery level
 scanning

---
 .../statusbar/policy/BluetoothControllerImpl.java  | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/policy/BluetoothControllerImpl.java b/packages/SystemUI/src/com/android/systemui/statusbar/policy/BluetoothControllerImpl.java
index abb0af0f57a62..7daade10c4b8b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/policy/BluetoothControllerImpl.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/policy/BluetoothControllerImpl.java
@@ -334,10 +334,18 @@ private void updateAudioProfile() {
 
     @Override
     public int getBatteryLevel() {
-        if (!mConnectedDevices.isEmpty()) {
-            return mConnectedDevices.get(0).getBatteryLevel();
+        synchronized (mConnectedDevices) {
+            if (mConnectedDevices.isEmpty()) {
+                return BluetoothDevice.BATTERY_LEVEL_UNKNOWN;
+            }
+            for (CachedBluetoothDevice device : mConnectedDevices) {
+                int level = device.getBatteryLevel();
+                if (level != BluetoothDevice.BATTERY_LEVEL_UNKNOWN) {
+                    return level;
+                }
+            }
+            return BluetoothDevice.BATTERY_LEVEL_UNKNOWN;
         }
-        return -1;
     }
 
     private void updateBattery() {
